libbeat/esleg/eslegclient/connection_integration_test.go

```diff
@@ -34,17 +33,25 @@ import (
 	"github.com/stretchr/testify/require"
 
 	"github.com/elastic/beats/v7/libbeat/esleg/eslegtest"
-	"github.com/elastic/beats/v7/libbeat/outputs"
 	conf "github.com/elastic/elastic-agent-libs/config"
 	"github.com/elastic/elastic-agent-libs/transport/httpcommon"
 )
 
 func TestConnect(t *testing.T) {
 	conn := getTestingElasticsearch(t)
-	err := conn.Connect()
+	err := conn.Connect(context.Background())
 	assert.NoError(t, err)
 }
 
+func TestConnectionCanBeClosedAndReused(t *testing.T) {
+	conn := getTestingElasticsearch(t)
+	ctx, cancel := context.WithCancel(context.Background())
+	assert.NoError(t, conn.Connect(ctx), "first connect must succeed")
+	assert.NoError(t, conn.Close(), "close must succeed")
+	cancel()
+	assert.NoError(t, conn.Connect(context.Background()), "calling connect after close must succeed")
+}
+
```

Comment:
just to double-check, this is the test to ensure the bug is fixed, right?

Reply:
Yes

Reply:
There are more tests as well: https://github.com/elastic/beats/blob/50a8b1ca3ac6099b7806726f2266b830c819e256/libbeat/tests/integration/elasticsearch_test.go